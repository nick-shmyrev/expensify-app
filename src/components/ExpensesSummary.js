import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import { getVisibleExpenses, getExpensesTotal } from '../store/selectors/expenses-selectors';

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => (
  <div>
    <p>Viewing {expensesCount} {expensesCount === 1 ? 'expense' : 'expenses'} totalling {numeral(expensesTotal / 100).format('$0,0.00')}</p>
  </div>
);

const mapStateToProps = (state) => {
  const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
  
  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: getExpensesTotal(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
