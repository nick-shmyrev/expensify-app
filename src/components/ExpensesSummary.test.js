import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from './ExpensesSummary';
import moment from 'moment';

describe('ExpensesSummary component', () => {
  test('should render correctly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={4213}/>);
    expect(wrapper).toMatchSnapshot();
  });
  
  test('should render correctly with 0 or 2+ expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={4213}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
