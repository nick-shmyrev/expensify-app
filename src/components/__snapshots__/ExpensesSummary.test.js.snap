// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpensesSummary component should render correctly with 0 or 2+ expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     
    expenses
     totalling 
    $42.13
  </p>
</div>
`;

exports[`ExpensesSummary component should render correctly with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $42.13
  </p>
</div>
`;
